buildtest/buildsystem/base.py

```diff
@@ -498,12 +502,14 @@ def run_tests(self, testfile):
         os.chdir(self._get_testdir())
         self.logger.debug(f"Changing to directory {self._get_testdir()}")
 
-        # Run the test file using the shell
-        cmd = [self.get_shell(), testfile]
-
-        self.logger.debug(f"Running Test via command: {' '.join(cmd)}")
+        # command to run the test
+        cmd = []
+        # build the run command that includes the shell path, shell options and path to test file
+        cmd += [self.shell["path"], self.shell["opts"], testfile]
```

Comment:
This again should reference properties and not dictionary items, given that you are interacting with the class.

Reply:
This is fixed now